From bbfbad161dc89198830b182fde3df29d5cc23f13 Mon Sep 17 00:00:00 2001
From: Cui GaoSheng <cuigaosheng1@huawei.com>
Date: Sun, 28 Mar 2021 23:13:00 -0400
Subject: [PATCH] arm32: kaslr: Add missing sections about relocatable
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: bbfbad161dc89198830b182fde3df29d5cc23f13


hulk inclusion
category: bugfix
bugzilla: 47952
CVE: NA

------------------------------------------------------------------------

Fix follow warnings:
armeb-linux-gnueabi-ld: warning: orphan section `.gnu.hash' from
`arch/arm/kernel/head.o' being placed in section `.gnu.hash'

Signed-off-by: Cui GaoSheng <cuigaosheng1@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm/kernel/vmlinux.lds.S | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/kernel/vmlinux.lds.S b/arch/arm/kernel/vmlinux.lds.S
index 1a9849968ce6..8af654bd16bf 100644
--- a/arch/arm/kernel/vmlinux.lds.S
+++ b/arch/arm/kernel/vmlinux.lds.S
@@ -70,6 +70,10 @@ SECTIONS
 #endif
 	_etext = .;			/* End of text section */
 
+	.gnu.hash : {
+		*(.gnu.hash)
+	}
+
 	RO_DATA(PAGE_SIZE)
 
 	. = ALIGN(4);
-- 
2.26.2

